### PR TITLE
Add SteamCMD

### PR DESCRIPTION
**Pull request by cprecioso**
The wrapper is because of #19999

### CODE REVIEW

File: Casks/steamcmd.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  # steamcdn-a.akamaihd.net was verified as official when first introduced to the cask
+  url 'https://steamcdn-a.akamaihd.net/client/installer/steamcmd_osx.tar.gz'
+  name 'SteamCMD'
+  homepage 'https://developer.valvesoftware.com/wiki/SteamCMD'
+  license :gratis
+
+  auto_updates true
+
+  binary '__cask_wrapper.sh', target: 'steamcmd'
```
Comment (vitorgalvao) [inline Casks/steamcmd.rb]: Why the `__`?

File: Casks/steamcmd.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  preflight do
+    set_permissions "#{staged_path}/steamcmd", '+x'
+
+    file = File.open "#{staged_path}/__cask_wrapper.sh", 'w'
+    file.puts '#!/usr/bin/env bash
+cd $(dirname $(readlink $_))
```
Comment (vitorgalvao) [inline Casks/steamcmd.rb]: Indentation must be kept. Follow [blender](https://github.com/caskroom/homebrew-cask/blob/master/Casks/blender.rb)’s example.

File: Casks/steamcmd.rb (hunk 1 above)
Comment (cprecioso) [inline Casks/steamcmd.rb]: Given that the downloaded tarball may be changed in the server with new files, I wanted to namespace the file just in case.

Comment (vitorgalvao) [inline Casks/steamcmd.rb]: Fair enough.

File: Casks/steamcmd.rb (hunk 2 above)
Comment (cprecioso) [inline Casks/steamcmd.rb]: Done

**Comment by cprecioso** [comment]
Wait, don't merge yet

**Comment by cprecioso** [comment]
There was an error in the wrapper script, arguments weren't being passed to the actual script. I fixed it now, it can be merged.

**Comment by vitorgalvao** [comment]
Thank you for the contribution. It was merged directly as ce6375caedf9b1cb02291027a703b05642f0d895 to keep commit history cleaner. Your contribution is still credited to you.

**Pull request closed without merging**